File: witterungsstation_py/bootstrap/app_bootstrap.py
from __future__ import annotations

from typing import Callable

from bootstrap.runtime_registrations import register_gui_runtime, runtime_factories
from core.context import AppContext
from model.app_config import load_runtime_config


def build_gui_context(*, view, log: Callable[[str], None], post: Callable[[Callable[[], None]], None]) -> AppContext:
    config = load_runtime_config()

    register_gui_runtime(view=view, log=log, post=post, config=config)
    service_factories, controller_factories = runtime_factories()

    state = {"config": config}
    return AppContext(
        state=state,
        service_factories=service_factories,
        controller_factories=controller_factories,
    )

File: frontend/src/bootstrap/appBootstrap.ts
import { AppContext } from '@/core/context'
import { getControllerFactories, getServiceFactories } from '@/core/runtimeRegistry'
import { createAppState, type AppState } from '@/app/appState'
import { registerRuntime } from './runtimeRegistrations'

export { createAppState }
export type { AppState }

export function buildAppContext(state?: AppState): AppContext {
  const appState = state || createAppState()

  registerRuntime()
  const serviceFactories = getServiceFactories()
  const controllerFactories = getControllerFactories()

  return new AppContext({
    state: appState,
    serviceFactories,
    controllerFactories,
  })
}
